src/cpu/aarch64/jit_sve_512_x8s8s32x_conv_kernel.cpp

```diff
@@ -0,0 +1,1480 @@
+            sub(reg_stack, reg_stack, 64);
+            str(vmm_tmp, ptr(reg_stack));
+            vmm_load_src(vmm_tmp, reg_addr, mask_flag);
+            zip1(vmm_tmp.b, vmm_tmp.b, vmm_tmp.b);
+            zip1(vmm_tmp.h, vmm_tmp.h, vmm_tmp.h);
+            sxtb(vmm.s, mask_all_one / T_m, vmm_tmp.s);
+            if (mask_flag) {
+                not_(mask_tmp.b, mask_all_one.b, ktail_mask.b);
+                mov(vmm.s, mask_tmp / T_m, 0);
+            }
+            ldr(vmm_tmp, ptr(reg_stack));
+            add(reg_stack, reg_stack, 64);
+            break;
+        case data_type::u8:
+            sub(reg_stack, reg_stack, 64);
+            str(vmm_tmp, ptr(reg_stack));
+            vmm_load_src(vmm_tmp, reg_addr, mask_flag);
+            zip1(vmm_tmp.b, vmm_tmp.b, vmm_tmp.b);
+            zip1(vmm_tmp.h, vmm_tmp.h, vmm_tmp.h);
+            uxtb(vmm.s, mask_all_one / T_m, vmm_tmp.s);
+            if (mask_flag) {
+                not_(mask_tmp.b, mask_all_one.b, ktail_mask.b);
+                mov(vmm.s, mask_tmp / T_m, 0);
+            }
+            ldr(vmm_tmp, ptr(reg_stack));
+            add(reg_stack, reg_stack, 64);
+            break;
+        default: assert(!"unsupported data type");
+    }
+    if (type_in != data_type::f32) scvtf(vmm_in.s, mask_all_one, vmm_in.s);
+}
+
+void jit_sve_512_x8s8s32x_fwd_kernel::store_output(
+        int ur_w, bool last_oc_block_flag) {
+    int nb_oc_block
+            = jcp.is_depthwise ? jcp.nb_ch_blocking : jcp.nb_oc_blocking;
+    int oc_block = jcp.is_depthwise ? jcp.ch_block : jcp.oc_block;
+
+    ldr(reg_bias, ptr(reg_param1, GET_OFF(bias)));
+    ldr(reg_ptr_scales, ptr(reg_param1, GET_OFF(scales)));
+    if (!jcp.signed_input)
+        ldr(reg_compensation, ptr(reg_param1, GET_OFF(compensation)));
+
+    const auto &p = attr_.post_ops_;
+    const int sum_idx = p.find(primitive_kind::sum);
+    const float *p_sum_scale = nullptr;
+    if (sum_idx != -1) {
+        const auto &p_entry = p.entry_[sum_idx];
+        p_sum_scale = &p_entry.sum.scale;
+    }
+
+    if (p_sum_scale && *p_sum_scale != 1.f)
+        mov_imm(reg_ptr_sum_scale, (size_t)p_sum_scale);
+
+    for (int k = 0; k < nb_oc_block; k++) {
+        const bool mask_flag
+                = last_oc_block_flag && k == nb_oc_block - 1 && mask_gflag;
+        int scale_offset = jcp.is_oc_scale * (sizeof(float) * k * oc_block);
+        if (jcp.with_bias) {
+            int bias_offset = jcp.typesize_bia * k * oc_block;
+
+            cvt2ps(jcp.bia_dt, vmm_bias, reg_bias, bias_offset, mask_flag);
+        }
+        if (!jcp.signed_input) {
+            int comp_offset = sizeof(int32_t) * k * oc_block;
+
+            cvt2ps(data_type::s32, vmm_comp, reg_compensation, comp_offset,
+                    mask_flag);
+        }
+        /* add to accum: compensation, bias and permute */
+        if (!jcp.is_fast_depthwise && jcp.signed_input) {
+            auto reg_addr = get_comp_addr_reg(reg_ptr_scales, scale_offset);
```

Comment:
I was wondering about the need of compensation related code for signed input case. If I understand correctly, `sdot` instruction works with `s8, s8` input and so we don't require any compensation.

Reply:
In the case of `s8`, `s8` input, `sdot` instruction works without compensation. We had retained register names used in the code for compensation. Therefore, we changed the register names.